jaas/src/main/java/com/griddynamics/jagger/jaas/service/impl/TestEnvironmentServiceImpl.java

```diff
@@ -100,7 +94,7 @@ else if (!isEqualCollection(testEnvToUpdate.getLoadScenarios(), newTestEnv.getLo
         }
 
         if (testEnvToUpdate.getRunningLoadScenario() != newTestEnv.getRunningLoadScenario()) {
-            testEnvToUpdate.setRunningLoadScenario(getNewRunningLoadScenario(newTestEnv, testEnvToUpdate));
+            testEnvToUpdate.setRunningLoadScenario(newTestEnv.getRunningLoadScenario());
```

Comment:
No. It's related to LoadScenarioEntity table. When you add new load scenario to test env and update it, this new load scenarion is persisted to db and obtains its id.
But when you set runningLoadScenario in the same transaction it's not persisted yet and exception will occur due to this. That's why you need to persist load scenario firstly by adding it to env and only after that you'll be able to set runningLoadScenario.